source/ada/lsp-ada_handlers.adb

```diff
@@ -3195,40 +3270,42 @@ package body LSP.Ada_Handlers is
 
       Emit_Progress_Begin;
 
-      for Context of Self.Contexts.Each_Context loop
-         for F in Context.List_Files loop
-            declare
-               File : constant GNATCOLL.VFS.Virtual_File :=
-                 LSP.Ada_File_Sets.File_Sets.Element (F);
-               URI  : constant LSP.Messages.DocumentUri := File_To_URI
-                 (LSP.Types.To_LSP_String (File.Display_Full_Name));
-            begin
-               if not Self.Open_Documents.Contains (URI) then
-                  Current_Percent := (Index * 100) / Total;
-                  --  If the value of the indexing increased by at least one
-                  --  percent, emit one progress report.
-                  if Current_Percent > Last_Percent then
-                     Emit_Progress_Report (Current_Percent);
-                     Last_Percent := Current_Percent;
-                  end if;
+      while not Self.Files_To_Index.Is_Empty loop
+         declare
+            Cursor : File_Sets.Cursor := Self.Files_To_Index.First;
```

Comment:
Minor note: We could get rid of Cursor altogether using

    File := Self.Files_To_Index.First_Element;
    Self.Files_To_Index.Delete_First;

Reply:
ah there is no First_Element or Delete_First for a Hashed_Set!